### PR TITLE
feat(TDP-5291): header bar react cmf

### PR DESCRIPTION
**Pull request by fabballe**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-5291

**Please check if the PR fulfills these requirements**
- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

File: dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { call, put, select } from 'redux-saga/effects';
+import api, { actions } from '@talend/react-cmf';
+import http from './http';
+import PreparationService from '../../services/preparation.service';
+
+const defaultHttpConfiguration = {
```
Comment (fabballe) [inline dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js]: switch this configuration inside `http.js` in order to have the same configuration for everybody

File: dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	const preparations = yield select(state => state.cmf.collections.get('preparations'));
+	const updated = preparations.update(
+		preparations.findIndex(val => val.get('id') === payload),
+		val => val.set('display', 'text'),
+	);
+	yield put(actions.collections.addOrReplace('preparations', updated));
+}
+
+export function* duplicate(prep) {
+	const newName = `test${Math.random()}`;
```
Comment (fabballe) [inline dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js]: Add TODO

File: dataprep-webapp/src/app/next/sagas/watchers/index.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { default as preparation } from './preparation.saga';
+import { default as bootstrap } from './bootstrap.saga';
+import { default as redirect } from './redirect.saga';
+import { default as help } from './help.saga';
+import { default as http } from './http.saga';
+
+
+export default {
+	redirect: Object.keys(redirect).map(k => redirect[k]),
```
Comment (fabballe) [inline dataprep-webapp/src/app/next/sagas/watchers/index.js]: unify syntax

File: dataprep-webapp/config/webpack.config.dev.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 const path = require('path');
 const webpack = require('webpack');
-const ReactCMFWebpackPlugin = require('@talend/react-cmf-webpack-plugin');
+// const ReactCMFWebpackPlugin = require('@talend/react-cmf-webpack-plugin');
```
[the remark this answers is not in the document]
Comment (ncomont) [reply dataprep-webapp/config/webpack.config.dev.js]: ✅

File: dataprep-webapp/config/webpack.config.dev.js [hunk 5]
```diff
@@ -19,9 +19,9 @@ config.plugins.push(
 	new webpack.DefinePlugin({
 		'process.env.NODE_ENV': JSON.stringify('development'),
 	}),
-	new ReactCMFWebpackPlugin({
-		watch: true,
-	}),
+	// new ReactCMFWebpackPlugin({
+	// 	watch: true,
+	// }),
```
[the remark this answers is not in the document]
Comment (ncomont) [reply dataprep-webapp/config/webpack.config.dev.js]: ✅

File: dataprep-webapp/src/app/configure.cmf.js [hunk 6]
```diff
@@ -1,25 +1,20 @@
...
 import actions from './next/actions';
 import components from './next/components/index';
 import App from './next/components/App.container';
-import {
-	ALERT,
-	FETCH_PREPARATIONS,
-} from './next/constants';
-import sagas from './next/sagas';
+import { ALERT } from './next/constants/actions';
+import { default as constants } from './next/constants';
```
[the remark this answers is not in the document]
Comment (ncomont) [reply dataprep-webapp/src/app/configure.cmf.js]: I removed the ALERT action

File: dataprep-webapp/src/app/configure.cmf.js [hunk 7]
```diff
@@ -1,25 +1,20 @@
...
+import sagas from './next/sagas/watchers';
+
+window.api = api;
+window.registry = api.registry.getRegistry();
```
[the remark this answers is not in the document]
Comment (ncomont) [reply dataprep-webapp/src/app/configure.cmf.js]: good catch

File: dataprep-webapp/src/app/configure.cmf.js [hunk 8]
```diff
@@ -122,31 +116,42 @@ export default function initialize(additionalConfiguration = {}) {
 		/**
 		 * Register action creators in CMF Actions dictionary
 		 */
-		registerActionCreator('preparation:fetchAll', actions.preparation.fetchAll);
 		registerActionCreator('preparation:duplicate', actions.preparation.duplicate);
 		registerActionCreator('preparation:edit:submit', actions.preparation.rename);
 		registerActionCreator('preparation:edit:cancel', actions.preparation.cancelRename);
-		registerActionCreator('preparation:open', actions.preparation.openPreparation);
+		registerActionCreator('preparation:open', actions.preparation.open);
```
[the remark this answers is not in the document]
Comment (ncomont) [reply dataprep-webapp/src/app/configure.cmf.js]: This is not supposed to be definitive. We're waiting for some CMF modifications.

File: dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js (hunk 2 above)
Comment (ncomont) [reply dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js]: ✅

File: dataprep-webapp/src/app/next/sagas/watchers/index.js (hunk 3 above)
Comment (ncomont) [reply dataprep-webapp/src/app/next/sagas/watchers/index.js]: ✅

File: dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js (hunk 1 above)
Comment (ncomont) [reply dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js]: ✅

**Pull request merged**